SLING-9612 Tune the nodeps javax.activation bundle to Java 8

bosschaert opened the pull request:
I also added the capabilities as described in Portable Java Contracts: https://www.osgi.org/portable-java-contract-definitions/

bosschaert committed:
SLING-9612 Tune the nodeps javax.activation bundle to Java 8

kwin commented on pom.xml:
```diff
@@ -58,8 +58,13 @@
+Bundle-Name: ${project.name} (Framework Fragment, No embedded dependencies)
 Bundle-SymbolicName: ${project.artifactId}.nodeps
+ExtensionBundle-Activator: org.apache.sling.javax.activation.internal.Activator
+Export-Package: javax.activation;version="0.0.0",com.sun.activation.registries;version="0.0.0",com.sun.activation.viewers;version="0.0.0"
```
This will embed. I would rather clean what is exported by the system bundle (https://lists.apache.org/thread.html/r0aa3c02c38b5026e24684a00de3c6c8506e166334e77892ac2ee0ab3%40%3Cdev.sling.apache.org%3E)

bosschaert replied:
Hi @kwin, my updated pr will not embed any more thanks to your link to https://groups.google.com/forum/#!topic/bndtools-users/NAdU0_X8OKY for the bnd instructions.

I agree that updating what is exported by the system bundle is a good idea. However I would still like to have the exports as defined by this PR in the bundle/fragment as I'd like to be able to run it with already released versions of Sling.

bosschaert committed:
Use -exportcontents instead of Export-Package

To avoid embedding the java.activation classes.
Also added the list of compatible JavaActivation contracts

kwin commented on pom.xml:
```diff
@@ -72,7 +77,9 @@ Bundle-SymbolicName: ${project.artifactId}.nodeps
                             <bnd><![CDATA[
 -includeresource:@jakarta.activation-[0-9.]*(-SNAPSHOT)?.jar
 # version automatically determined from the MANIFEST.MF of the jakarta.activation.jar
+Bundle-Activator: org.apache.sling.javax.activation.internal.Activator
```
Why was the annotation from https://github.com/apache/sling-org-apache-sling-javax-activation/pull/4/files#diff-647f7f295a3a772b13eaaa75470004f7L52 not working?

bosschaert replied:
The nodeps variant is a framework extension fragment. That one needs a different header: `ExtensionBundle-Activator` instead of `Bundle-Activator`, so I couldn't use the annotation for the 2 ways in which the activator was used.

The actual interface to be implemented by the Extension Bundle Activator is the same as the normal Bundle Activator.

bosschaert commented:
Thanks for the review, @kwin !
